src/main/java/at/hannibal2/skyhanni/features/garden/farming/LaneswitchNotification.kt

```diff
@@ -0,0 +1,76 @@
+package at.hannibal2.skyhanni.features.garden.farming
+
+import at.hannibal2.skyhanni.config.features.garden.laneswitch.LaneswitchFarmConfig
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.features.garden.GardenAPI
+import at.hannibal2.skyhanni.utils.LocationUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.sendTitle
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SoundUtils
+import at.hannibal2.skyhanni.utils.SoundUtils.playSound
+import kotlin.time.Duration.Companion.seconds
+
+class LaneswitchNotification {
+
+    private val config get() = GardenAPI.config.laneswitch
+
+    private var blocksPerSecond = 0.0
+    private var lastPos = LorenzVec(0, 0, 0)
+    private var lastLaneSwitch = SimpleTimeMark.farPast()
+
+    private fun coverable(from: LorenzVec, to: LorenzVec, speed: Double, time: Int): Boolean {
+        return from.distance(to) <= speed * time
+    }
+
+    companion object {
+        private val config get() = GardenAPI.config.laneswitch
+
+        @JvmStatic
+        fun playUserSound() {
+            SoundUtils.createSound(
+                config.notification.sound.notificationSound,
+                config.notification.sound.notificationPitch,
+                config.notification.sound.notificationVolume
+            ).playSound()
+        }
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!config.enabled || !GardenAPI.inGarden()) return
+        if (GardenCropSpeed.averageBlocksPerSecond <= 0.0 && config.notification.farmingOnly) return
+
+        val playerPosition = LocationUtils.playerLocation().round(2)
+        blocksPerSecond = playerPosition.distance(lastPos) * 20
+        this.lastPos = playerPosition
+
+        if (lastLaneSwitch.passedSince() >= config.notification.settings.notificationTimeout.seconds) {
+            val farmEnd: List<LorenzVec> = when (config.farm.farmDirection) {
+                LaneswitchFarmConfig.FarmDirection.NORTH, LaneswitchFarmConfig.FarmDirection.SOUTH -> {
+                    val xValue = (config.farm.plotAmount * 48).toDouble()
+                    listOf(
+                        LorenzVec(xValue, playerPosition.y, playerPosition.z),
+                        LorenzVec(-xValue, playerPosition.y, playerPosition.z)
+                    )
+                }
+
+                LaneswitchFarmConfig.FarmDirection.EAST, LaneswitchFarmConfig.FarmDirection.WEST -> {
+                    val zValue = (config.farm.plotAmount * 48).toDouble()
+                    listOf(
+                        LorenzVec(playerPosition.x, playerPosition.y, zValue),
+                        LorenzVec(playerPosition.x, playerPosition.y, -zValue)
+                    )
+                }
+
+                else -> emptyList()
```

Comment:
Why is this here? This should be automatically covered since enums can be exhaustively covered. If this is meant as a graceful null check, it fails, since the next line will result in an AIOBE. If you must explicitly check for null, just throw here so that it gets caught by the normal error handler with a meaningful error message. Otherwise just let normal kotlin null handling handle it.

Reply:
= "Just remove this line, it is fine/better this way"

Reply:
Are you talking about the else -> emptyList?

Reply:
Yes, we are talking about that. You see the line we are talking about directly above the comment here:
![image](https://github.com/hannibal002/SkyHanni/assets/24389977/aefd4f06-7ae2-494a-9391-896fbd5949ff)